src/ucp/core/ucp_context.c

```diff
@@ -1037,6 +1046,25 @@ ucs_status_t ucp_init_version(unsigned api_major_version, unsigned api_minor_ver
         goto err_free_ctx;
     }
 
+    if (config->version_error_mode != UCP_VERSION_CHECK_IGNORE) {
+        if ((api_major_version != major_version) || (api_minor_version != minor_version)) {
```

Comment:
1) seems like it should be in separate function.
2) i think it makes sense don't ignore major version mismatch, at least need to print warning.

Reply:
It makes sense to check in the very beginning, like it was done before. Why would you need to allocate and fill context if UCX version is invalid?

Reply:
@brminich we need config initialized prior to version check

Reply:
maybe would be better to have different check:
version would be considered compatible, if (api_minor_version <= minor_version), and major has not changed (since minot versions are backward compat)
if version is not compatible - print a warning.
no config vars.